addons/l10n_ke_edi_oscu/models/account_tax.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    l10n_ke_tax_type_id = fields.Many2one(
+        'l10n_ke_edi_oscu.code',
+        string='KRA Tax Code',
+        domain=[('code_type', '=', '04')],
+        compute='_compute_l10n_ke_tax_type_id',
+        store=True, readonly=False,
+        help='KRA code that describes a tax rate or exemption.',
+    )
+
+    @api.depends('amount')
+    def _compute_l10n_ke_tax_type_id(self):
```

Comment:
This compute is there just for user-friendliness, to auto-set the code for the 16% and 8% tax rates (where there is only one possible tax code). 

However, the user can still change this field manually, and will need to set it themselves for 0% taxes since in that case we can't automatically guess whether we are dealing with a 0% VAT or a VAT Exempt tax.